transaction.go

```diff
@@ -59,10 +59,42 @@ func (t *Transaction) SetScript(script []byte) *Transaction {
 
 // AddArgument adds a Cadence argument to this transaction.
 func (t *Transaction) AddArgument(arg cadence.Value) *Transaction {
+	encodedArg, err := jsoncdc.Encode(arg)
+	if err != nil {
+		// argument is not added if encoding fails
```

Comment:
no failure signal is returned here. Is this to keep the interface backward compatible?

Reply:
Yeah, that was my original intent, but I'm having second thoughts. I think this should be more explicit about the possibility of an error.

@Kay-Zee @turbolent which API seems best to you? I leaning towards option D. Even though its probably the worst UX, it's by far the most explicit. 

**Option A**
```go
var myArg cadence.Value

tx.AddArgument(myArg) // no error, fails silently

tx.AddRawArgument(jsoncdc.MustEncode(myArg))
```

**Option B**
```go
var myArg cadence.Value

tx.AddArgument(myArg) // panics if encoding fails

tx.AddRawArgument(jsoncdc.MustEncode(myArg))
```

**Option C**
```go
var myArg cadence.Value

err := tx.AddArgument(myArg) // error for failed encoding, no chainability

tx.AddRawArgument(jsoncdc.MustEncode(myArg))
```

**Option D**

Caller handles all encoding

```go
var myArg cadence.Value

// AddArgument accepts []byte, encoding must be done by caller
// Note: AddRawArgument(b []byte) and Argument(i int) functions removed
tx.AddArgument(jsoncdc.MustEncode(myArg))
```

Reply:
I'm actually more partial to option B, with a name change:

```go
var myArg cadence.Value

tx.MustAddArgumentValue(myArg) // panics if encoding fails

tx.AddArgument(jsoncdc.MustEncode(myArg))
```

with `MustAddArgumentValue` being just a slice convenience of not having to import `jsoncdc` for the `MustEncode` function. The naming change will also make having the user handle the encoding seem more like the default here, which might actually be what we want.

I'm overall most worried about failing silently, since i know it's not really suppose to happen, but still...

Reply:
Another thing I just thought of is the fact that we may want to swap the argument encoding format in the future, and therefor it could be good to keep JSON-CDC completely separate from the `flow.Transaction` implementation.

Seeing as your proposal builds on top of option D, how would you feel about merging option D today and then adding the `MustAddArgumentValue` as a convenience function later if there ends up being demand for it?

Reply:
I think it's very likely and desirable that we use a more efficient binary encoding format for the arguments soon, so it might be a good idea to keep it flexible. However, this would be additive and we would need a way to differentiate (tag) the encoded data.

I don't really see a problem of having the SDK know about supported encoding formats and providing the convenience of performing the best encoding given a `cadence.Value` (one could even imagine it's based on a "debug mode" flag that uses the more human-readable JSON format instead of the more efficient binary format)

Reply:
After giving this more thought and trying out some sample code, I think I like option C the best. Although the error return breaks chain-ability of the setter method, I think it's still the best combination of correctness (error is not implicit or hidden) and convenience for the users.

@Kay-Zee @turbolent Do you mind taking another look?